guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/context/GuvnorStructureContextBranchChangeHandler.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.structure.client.editors.context;
+
+
+import org.guvnor.structure.repositories.Repository;
+
+public interface GuvnorStructureContextBranchChangeHandler {
```

Comment:
Yes. I could use events. But I find that events are so often miss used. If I make a ChangeBranchEvent here then next time I'll find it from somewhere in the server side. Not saying we should not use Events, but this is a context for Admin Perspective, so this felt safer. Also the event will more likely leak out of Admin Perspective.